From 50487bf64f3f120ee4ef46ef4af73903fc4d24bf Mon Sep 17 00:00:00 2001
Message-Id: <50487bf64f3f120ee4ef46ef4af73903fc4d24bf.1429601425.git.chang-joon.lee@intel.com>
In-Reply-To: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
References: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Thu, 16 Apr 2015 22:54:14 +0530
Subject: [PATCH 4/9] REVERTME! [VPG]: drm/i915: Use default DSPARB values.

This patch writes default fixed dsparb values for hdmi.

REVERTME: This patch is revertme as the right method would
be to update the calculated dsparb values even incase of hdmi.
This has to be reverted once the patch that updates the calulated
values comes in.

Issue: GMINL-4468
Change-Id: I60f47754b8372abed233a04a6df028d63a52d01b
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index d6cc4d7..0b72623 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1595,6 +1595,8 @@ void vlv_update_dsparb(struct intel_crtc *intel_crtc)
 	int pipe_stat = VLV_PIPE_STATS(dev_priv->pipe_plane_stat);
 	u32 dsparb = 0;
 	u32 dsparb_h = 0;
+	u32 dsparb3 = DSPARB_50_25_25;
+	u32 dsparb2 = DSPARB2_50_25_25;
 	u32 pa = 0, sa = 0, sb = 0;
 	int fifo_size = 0;
 
@@ -1706,8 +1708,8 @@ void vlv_update_dsparb(struct intel_crtc *intel_crtc)
 
 		break;
 	case PIPE_C:
-		I915_WRITE_BITS(DSPARB3, dsparb, DSPARB3_PIPEC_MASK);
-		I915_WRITE_BITS(DSPARB2, (dsparb_h << DSPARB2_PIPEC_SHIFT),
+		I915_WRITE_BITS(DSPARB3, dsparb3, DSPARB3_PIPEC_MASK);
+		I915_WRITE_BITS(DSPARB2, (dsparb2 << DSPARB2_PIPEC_SHIFT),
 						DSPARB2_PIPEC_MASK);
 		/* update wm */
 		I915_WRITE_BITS(DSPFW9, (pa << VLV_FW_PIPEC_PC_SHIFT),
-- 
1.7.9.5

